pkg/workloads/reservation.go

```diff
@@ -172,6 +173,11 @@ func (a *API) create(r *http.Request) (interface{}, mw.Response) {
 		return ReservationCreateResponse{ID: id}, mw.PaymentRequired(errors.New("pool needs additional capacity to support this workload"))
 	}
 
+	err = a.checkAllowedPublicIPs(workload, db)
```

Comment:
I think a proper name for this method is `allocatePublicIP` since a `check` means it's a check so it validates something and not actually creating any side-effect, unlike what this method actually does.